libgalois/src/analytics/matrix_completion/matrix_completion.cpp

```diff
@@ -0,0 +1,534 @@
+                    millis / 1e6;
+
+    int curRound = round + deltaRound;
+    katana::gPrint(
+        "R: ", curRound, " elapsed (ms): ", curElapsed, " GFLOP/s: ", gflops);
+    if (plan.useExactError()) {
+      katana::gPrint(
+          " RMSE (R ", curRound, "): ", std::sqrt(error / graph.num_edges()), "\n");
+    } else {
+      katana::gPrint(
+          " Approx. RMSE (R ", (curRound - 1),
+          ".5): ", std::sqrt(std::abs(error / graph.num_edges())), "\n");
+    }
+
+    katana::gPrint("Error Change : ", std::abs((last - error) / last), "\n");
+    if (!isFinite(error))
+      break;
+    if (plan.fixedRounds() <= 0 &&
+        (round >= plan.maxUpdates() || std::abs((last - error) / last) < plan.tolerance()))
+      break;
+    if (sf.isBold()) {
+      // Assume that loss decreases first round
+      if (last >= 0.0 && last < error)
+        rate = steps[deltaRound - 1] * 0.5;
+      else
+        rate = steps[deltaRound - 1] * 1.05;
+    }
+    last = error;
+  }
+}
+
+
+class SGDItemsAlgo {
+
+public:
+  bool isSgd() const { return true; }
+
+  std::string name() const { return "sgdItemsAlgo"; }
+
+  size_t numItems() const { return NUM_ITEM_NODES; }
+
+private:
+  struct Execute {
+    Graph& graph;
+    katana::GAccumulator<unsigned>& edges_visited;
+
+    void operator()(
+        LatentValue* steps, int, katana::GAccumulator<double>* errorAccum, MatrixCompletionPlan plan) {
+      const LatentValue step_size = steps[0];
+      katana::do_all(
+          katana::iterate(graph.begin(), graph.begin() + NUM_ITEM_NODES), [&](GNode src) {
+            for (auto ii : graph.edges(src)) {
+              auto dst = graph.GetEdgeDest(ii);
+              auto& item_latent_vector = graph.GetData<NodeLatentVector>(src);
+              auto& user_latent_vector = graph.GetData<NodeLatentVector>(dst);
+              LatentValue error = doGradientUpdate<NodeLatentVector>(item_latent_vector, user_latent_vector,
+                  plan.lambda(), graph.GetEdgeData<EdgeWeight>(ii),
+                  step_size);
+
+              edges_visited += 1;
+              if (plan.useExactError())
+                *errorAccum += error;
+            }
+          },
+          katana::loopname("sgdItemsAlgo"));
+    }
+  };
+public:
+  void operator()(Graph& graph, const StepFunction& sf, MatrixCompletionPlan plan) {
+    katana::GAccumulator<unsigned> edgesVisited;
+
+    katana::StatTimer executeTimer("Time");
+    executeTimer.start();
+
+    Execute fn{graph, edgesVisited};
+    executeUntilConverged(sf, graph, fn, plan);
+
+    executeTimer.stop();
+
+    katana::ReportStatSingle(
+        "sgdItemsAlgo", "EdgesVisited", edgesVisited.reduce());
+  }
+};
+
+
+template <typename Algo>
+katana::Result<void>
+Run(katana::PropertyGraph* pg, MatrixCompletionPlan plan) {
+  
+  auto result = ConstructNodeProperties<NodeData>(pg);
+  if (!result) {
+    KATANA_LOG_FATAL("failed to construct node properties: {}", result.error());
```

Comment:
All calls to `KATANA_LOG_FATAL` and `KATANA_DIE` need to replaced with Error returns.

Reply:
In cases like this you can just use `KATANA_CHECKED`.